### ISSUE TITLE
Issue with continued training from saved model

### ISSUE THREAD

**Issue by rai-harshit** [amaiya/ktrain#357]
First of all, thanks a lot for building this AWESOME library. This has saved me a lot of time and efforts.
Here is the issue I'm facing:
I finetuned the DistlibBERT model using the following piece of code:

```
model_ = t.get_classifier()
learner = ktrain.get_learner(model, train_data=trn, val_data=val, batch_size=4)
learner.fit_onecycle(5e-5, 5)
```
After 5 epochs, I had:
![image](https://user-images.githubusercontent.com/26897848/115059226-dd36b380-9f03-11eb-97ee-b8d81532ed63.png)

After this, I saved this fine-tuned model using the following piece of code:
```
learner.model.save_pretrained('../results/models/bert_ktrain/160421_ktrain_balanced_processed_8k_10e/')
```
Everything went pretty well till now.
Now, I load the fine-tuned model using the following piece of code (which has been mentioned in the FAQs as well):
```
from transformers import *
model = TFAutoModelForSequenceClassification.from_pretrained('../results/models/bert_ktrain/160421_ktrain_balanced_processed_8k_5e/')
model.compile(loss='categorical_crossentropy',optimizer='adam', metrics=['accuracy'])
learner = ktrain.get_learner(model, train_data=trn, val_data=val,batch_size=4)
learner.fit_onecycle(5e-5, 5)
```

Now, here is the problem:
When the training starts, the loaded model has an extremely high loss and the training accuracy continues to fall as the first epoch moves ahead.
![load_saved_model](https://user-images.githubusercontent.com/26897848/115059904-d2305300-9f04-11eb-9d83-855d2754b23a.PNG)

Now, as you can conclude from the two screenshots, the training accuracy should be around 94-95% and loss around 0.1-0.2 for the 6th epoch which is certainly not the case.
Can you help me here in understanding what exactly went wrong and what could be done to fix this?

Thanks :)

**Comment by amaiya** [comment]
It might be because your dataset has more than two classes (which is the default for `from_pretrained` I think).  If you have more than two classes, you'll need to specify this as an argument in the call to `from_pretrained`.  Please see the **transformers** documentation for more information on how to do this. (I've updated the FAQ to also mention this.)

I would recommend you just use [Method 1 from the FAQ](https://github.com/amaiya/ktrain/blob/master/FAQ.md#how-do-i-resume-training-from-a-saved-checkpoint), which automatically does this for you.  If you resume training by doing the following, I think you'll see the correct behavior.

```python
# save Predictor (i.e., model and Preprocessor instance) after partially training
ktrain.get_predictor(model, preproc).save('/tmp/my_predictor')

# reload Predictor and extract model
model = ktrain.load_predictor('/tmp/my_predictor').model

# re-instantiate Learner and continue training
learner = ktrain.get_learner(model, train_data=trn, val_data=val)
learner.fit_onecycle(2e-5, 1)
```
